lib/AdobeMarketingCloud/Api/Report.php

```diff
@@ -0,0 +1,183 @@
+<?php
+
+namespace AdobeMarketingCloud\Api;
+
+/**
+ * Api calls for getting data about reports.
+ *
+ * @link      https://marketing.adobe.com/developer/documentation/analytics-reporting-1-4/get-started
+ * @author    Brent Shaffer <bshafs at gmail dot com>
+ * @license   MIT License
+ */
+class Report extends SuiteApi
+{
+    /**
+     * Retrieve the elements to call from your report suite
+     *
+     * @param string $reportSuiteID
+     * @param bool $returnAsIdArray
+     *
+     * @return array
+     */
+    public function getElements($reportSuiteID, $returnAsIdArray = false)
+    {
+        $response = $this->post('Report.GetElements', array('reportSuiteID' => $reportSuiteID));
+        if ($returnAsIdArray) {
+            $filtered = array();
+            foreach ($response as $element) {
+                $filtered[$element['id']] = $element['name'];
+            }
+            return array_unique($filtered);
+        }
+        return $this->returnResponse($response);
+    }
+
+    /**
+     * Retrieve events for your report suites
+     *
+     * @param string $reportSuiteID
+     * @param bool $returnAsIdArray
+     *
+     * @return array
+     */
+    public function getMetrics($reportSuiteID, $returnAsIdArray = false)
+    {
+        $response = $this->post('Report.GetMetrics', array('reportSuiteID' => $reportSuiteID));
+        if ($returnAsIdArray) {
+            $filtered = array();
+            foreach ($response as $metric) {
+                $filtered[$metric['id']] = $metric['name'];
+            }
+            return array_unique($filtered);
+        }
+        return $this->returnResponse($response);
+    }
+
+    /**
+     * Submits a report request.
+     *
+     * https://marketing.adobe.com/developer/documentation/analytics-reporting-1-4/r-queue
+     *
+     * @param   array $reportDescription  the report description array
+     * @return  array
+     *   - reportID: The report ID of the queued report
+     */
+    public function queue($reportDescription)
+    {
+        $response = $this->post('Report.Queue', array(
+            'reportDescription' => $reportDescription,
+        ));
+
+        return $this->returnResponse($response);
+    }
+
+    /**
+     * Gets a report by report ID
+     * https://marketing.adobe.com/developer/documentation/analytics-reporting-1-4/r-get
+     *
+     * @param   mixed     The report ID or report description
+     * @return  tns:report      A structure containing the report data for the specified reportID
+     */
+    public function getReport($report)
+    {
+        if (is_numeric($report)) {
+            $response = $this->post('Report.Get', array(
```

Comment:
get() doesn't work because it is used in the base class as an http operation. I've converted them to { get | queue | cancel }Report()